platform/entanglement/src/services/CopyService.js

```diff
@@ -54,45 +57,156 @@ define(
             );
         };
 
-        CopyService.prototype.perform = function (domainObject, parent) {
-            var model = JSON.parse(JSON.stringify(domainObject.getModel())),
+        /**
+         * Will build a graph of an object and all of its child objects in
+         * memory
+         * @param domainObject The original object to be copied
+         * @param parent The parent of the original object to be copied
+         * @returns {Promise} resolved with an array of clones of the models
+         * of the object tree being copied. Copying is done in a bottom-up
+         * fashion, so that the last member in the array is a clone of the model
+         * object being copied. The clones are all full composed with
+         * references to their own children.
+         */
+        CopyService.prototype.buildCopyPlan = function(domainObject, parent, progress) {
```

Comment:
I'd mark this `@private`, don't think it makes sense outside of the broader copying workflow.